libraries/lib-wx-init/SettingsWX.cpp

```diff
@@ -0,0 +1,226 @@
+      while(mConfig->GetNextGroup(group, index))
+         groups.push_back(group);
+      return groups;
+   }
+   return {};
+}
+
+wxArrayString SettingsWX::GetChildKeys() const 
+{
+   long index;
+   wxString key;
+   if(mConfig->GetFirstEntry(key, index))
+   {
+      wxArrayString keys;
+      keys.push_back(key);
+      while(mConfig->GetNextEntry(key, index))
+         keys.push_back(key);
+      return keys;
+   }
+   return {};
+}
+
+bool SettingsWX::HasEntry(const wxString& key) const
+{
+   return mConfig->HasEntry(MakePath(key));
+}
+
+bool SettingsWX::HasGroup(const wxString& key) const
+{
+   return mConfig->HasGroup(MakePath(key));
+}
+
+bool SettingsWX::Remove(const wxString& key)
+{
+   if(key.empty())
+   {
+      {
+         const auto children = GetChildGroups();
+         for(auto& group : children)
+            mConfig->DeleteGroup(group);
+      }
+      {
+         const auto children = GetChildKeys();
+         for(auto& entry : children)
+            mConfig->DeleteEntry(entry, false);
+      }
+      return true;
+   }
+   const auto path = MakePath(key);
+   if(mConfig->HasEntry(path))
+      return mConfig->DeleteEntry(path, false);
+   if(mConfig->HasGroup(path))
+      return mConfig->DeleteGroup(path);
+   return false;
+}
+
+void SettingsWX::Clear()
+{
+   mConfig->DeleteAll();
+}
+
+bool SettingsWX::Read(const wxString& key, bool* value) const
+{
+   return mConfig->Read(MakePath(key), value);
+}
+
+bool SettingsWX::Read(const wxString& key, int* value) const
+{
+   return mConfig->Read(MakePath(key), value);
+}
+
+bool SettingsWX::Read(const wxString& key, long* value) const
+{
+   return mConfig->Read(MakePath(key), value);
+}
+
+bool SettingsWX::Read(const wxString& key, long long* value) const
+{
+   wxString str;
+   if(mConfig->Read(MakePath(key), &str))
+   {
+      if(str.ToLongLong(value))
+         return true;
+   }
+   return false;
+}
+
+bool SettingsWX::Read(const wxString& key, double* value) const
+{
+   return mConfig->Read(MakePath(key), value);
+}
+
+bool SettingsWX::Read(const wxString& key, wxString* value) const
+{
+   return mConfig->Read(MakePath(key), value);
+}
+
+bool SettingsWX::Write(const wxString& key, bool value)
+{
+   return mConfig->Write(MakePath(key), value);
+}
+
+bool SettingsWX::Write(const wxString& key, int value)
+{
+   return mConfig->Write(MakePath(key), value);
+}
+
+bool SettingsWX::Write(const wxString& key, long value)
+{
+   return mConfig->Write(MakePath(key), value);
+}
+
+bool SettingsWX::Write(const wxString& key, long long value)
+{
+   return mConfig->Write(MakePath(key), wxString::Format("%lld", value));
+}
+
+bool SettingsWX::Write(const wxString& key, double value)
+{
+   return mConfig->Write(MakePath(key), value);
+}
+
+bool SettingsWX::Write(const wxString& key, const wxString& value)
+{
+   return mConfig->Write(MakePath(key), value);
+}
+
+bool SettingsWX::Flush() noexcept
+{
+   try
+   {
+      return mConfig->Flush();
```

Comment:
Probably it's worth commenting that complex Flush implementation is hidden in mConfig now.